embassy-mspm0:  add CPU accelerated float division function

bespsm opened the pull request:
- mspm0-mathacl: add division example
- mspm0-mathacl: add CPU accelerated float division function
- using internal Mathacl Qval type for providing length of fractional part

i509VCB commented:
From what I can tell, the MathACL does:
- i32
- u32
- u16f16
- i16f16

For DIV anything from i1f31 to i32 and vice versa for signed?

bespsm commented:
@i509VCB  I hope I understood your comment correctly. I added IQType, where the integer and the fractional parts are set explicitly, also added division for i32 and u32. The examples are adopted

i509VCB reviewed (approved):
I'm no expert with regards to this part of the hardware, but otherwise it looks fine.

Edit: Please squash if possible so I can merge.
